### PR TITLE
#6 Add transaction requirements

### LINKED ISSUE #6: Outline requirements for Transaction functionality
## Description
Based on the [Python code](https://github.com/Zigurat-blockchain-masters/blockchain-core-python-draft) provided by Zigurat, we need to analyze requirements what do we need to do about Transaction functionality and capture them as requirements for future implementation.

Follow repository https://github.com/Zigurat-blockchain-masters/blockchain-core-python-draft for source code and other details

## Acceptance criteria

1. _Requirements are captured for the Transaction functionality from Python code:_
    1. UTXO

1. Pull Request is created against `functional-spec-srs` branch
1. Functional specification file updated with a proper markdown formatting (following the same standard as currently present in that file). See https://github.com/Zigurat-blockchain-masters/blockchain-core/blob/functional-spec-srs/docs/Functional%20Documentation/Functional_Specification.md

### PR DESCRIPTION
**Pull request by JCRigo86**
### Description

Completes the functional specification for Transaction requirements

### Related Issue

Fixes #6 

### Changes Made

1. Added paragraph 1.4 about transaction requirements 
2. Updated Blocks Error Handling
3. Added UTXO definition

### How to Test

Not applicable

### By submitting this pull request, I confirm the following (please tick checkboxes):

- [x] Any new functions, methods, or classes are properly documented.
- [x] The commit messages are clear and descriptive.
- [x] The pull request title is concise and descriptive.
- [x] The pull request is targeted to the correct branch.

### CODE REVIEW

File: docs/Functional Documentation/Functional_Specification.md [hunk 1]
```diff
@@ -143,21 +144,171 @@ The codebase will be written in JavaScript as per decision made in [ADR-001](htt
...
+   - Input: acts on UTXO
+   - Output: Dictionary (originally: Python format)
+   - Purpose: This method returns a dictionary containing the UTXO's attributes (tx_hash, public_key, and message) as key-value pairs. The method creates the dictionary using the current instance's attribute values.
+   - Error handling: None
+
+2. `Get Hash`
+   - Input: acts on UTXO
+   - Output: hashed value of UTXO's dictionary
+   - Purpose: This method calculates a hash value for the dictionary representation of the UTXO. It calls the Get Dictionary (see 1. above) method to obtain the UTXO's attribute dictionary. It then converts the dictionary to a JSON string using json.dumps. The resulting JSON string is passed to the hashing.hash function (provided by the hashing module) to compute a hash value. The calculated hash value is returned as the result of the Get Hash method.
+   - Error handling: None
```
Comment (rkokhatskyi) [inline docs/Functional Documentation/Functional_Specification.md:196]: hey @JCRigo86, sounds like we should have some error handling here and in other places as well

**Review by Agn21 — APPROVED** [review]

File: docs/Functional Documentation/Functional_Specification.md (hunk 1 above)
Comment (JCRigo86) [reply docs/Functional Documentation/Functional_Specification.md:196]: hey @rkokhatskyi reviewing this spec, all methods here except one (Sign method in "unsigned transaction" class) actually act on "self" and not on inputs. I've updated the Error handling for that Sign method, however I'm not sure which error handling to set in the other methods. Can we consider it as solved until we have better visibility on how the code integrates overall? Unless you have suggestions on those error handlings

Comment (rkokhatskyi) [reply docs/Functional Documentation/Functional_Specification.md:196]: thanks @JCRigo86, let's close this off therefore 👍

**Review by rkokhatskyi — APPROVED** [review]
LGTM 👍

**Pull request merged**